File: code/ch07/7.2.1_get_sum.cpp
class Solution {
public:
    int getSum(int a, int b) {
        int ans = a ^ b;
        int c = a & b;
        while(c != 0) {
            c = (c & 0xffffffff) << 1;
            int ans_prim = ans ^ c;
            c = ans & c;
            ans = ans_prim;
        }
        return ans;
    }
};

File: code/ch07/7.2.1_get_sum.py
class Solution:
    def getSum(self, a:int, b:int)->int:
        carry = 0
        mask  = 0xFFFFFFFF
        
        while b & mask != 0:
            carry = (a & b) << 1
            a = a ^ b
            b = carry
            
        if b > mask:
            return a & mask
        else:
            return a
